### ISSUE TITLE
clear source labels and annotations on odigos uninstall

### ISSUE THREAD

**Issue by blumamir** [keyval-dev/odigos#665]
When a user runs `odigos uninstall` from cli, there are various cleanups that take place.

I noticed that one thing that is currently not taken care of is the cleanup of the odigos labels and annotations from k8s Deployment/StatefulSet/DaemonSet.

We should apply it [here](https://github.com/keyval-dev/odigos/blob/c31ae9804bc49f1994cb248728912d660c33c204/cli/cmd/uninstall.go#L103). Basically, the same way as we set annotations [here for example](https://github.com/keyval-dev/odigos/blob/c31ae9804bc49f1994cb248728912d660c33c204/frontend/endpoints/sources.go#L105) and set label [here](https://github.com/keyval-dev/odigos/blob/c31ae9804bc49f1994cb248728912d660c33c204/frontend/endpoints/sources.go#L215), they should be reverted when odigos is uninstalled.

To check your fix, you can set up an odigos environment with [Quickstart](https://docs.odigos.io/intro). Then, make necessary changes in the cli module code, and run in terminal:

```
➜  odigos git:(main) cd cli
➜  cli git:(main) go run . uninstall
```
To run the cli uninstall command locally from code and verify your fix.

**Comment by blumamir** [comment]
Hi @georgettica 
Thanks

Please reach out here or in odigos slack if you have any questions or need any assitence

**Comment by miravtmehta** [comment]
@blumamir While deleting all the objects of _Deployment/StatefulSet/DaemonSet_, When we uninstall CLI, it delete's all the objects under namespace **odigos-system** 

so why do need to explicitly delete annotations and labels ? Am i missing something ?

**Comment by blumamir** [comment]
> @blumamir While deleting all the objects of _Deployment/StatefulSet/DaemonSet_, When we uninstall CLI, it delete's all the objects under namespace **odigos-system**
> 
> so why do need to explicitly delete annotations and labels ? Am i missing something ?

Odigos installs it's own objects under the `odigos-system` namespace which are currently deleted with uninstall, but it also modifies the existing user objects (Deployment/StatefulSet/DaemonSet) which can reside in any namespace. 

[here](https://github.com/keyval-dev/odigos/blob/c31ae9804bc49f1994cb248728912d660c33c204/frontend/endpoints/sources.go#L105) and [here](https://github.com/keyval-dev/odigos/blob/c31ae9804bc49f1994cb248728912d660c33c204/frontend/endpoints/sources.go#L215) you can see how they are being applied to a k8s object in an arbitrary namespace.

**Comment by sachin2605** [comment]
@blumamir I am able to get what the issue is.  So we should be deleting the odigos annotations from all the resources in any namespaces. I can work on this and you can assign to me.

**Comment by tomasmota** [comment]
Hey @blumamir , it seems like this has been fixed by #854? https://github.com/keyval-dev/odigos/blob/9d07135a5ad2d64ab9658e9626d4fe7b49ce7397/cli/cmd/uninstall.go#L123

**Comment by blumamir** [comment]
> Hey @blumamir , it seems like this has been fixed by #854?
> 
> https://github.com/keyval-dev/odigos/blob/9d07135a5ad2d64ab9658e9626d4fe7b49ce7397/cli/cmd/uninstall.go#L123

indeed.
Thanks for updating here 🙏
